test/functional/test_framework/ellsq.py

```diff
@@ -0,0 +1,243 @@
+# Source: https://github.com/sipa/writeups/tree/main/elligator-square-for-bn
+"""Test-only Elligator Squared implementation
+
+WARNING: This code is slow and uses bad randomness.
+Do not use for anything but tests."""
+
+import random
+import secrets
+import unittest
+
+from .key import fe, SECP256K1, SECP256K1_FIELD_SIZE, SECP256K1_G, SECP256K1_ORDER
+
+C1 = fe(-3).sqrt()
+C2 = (C1 - fe(1)) / fe(2)
+B = fe(7)
+
+def f(u):
```

Comment:
Maybe give this function a more descriptive name, and a longer comment?

E.g. "Given any field element u (of type fe), return the affine X and Y coordinates of a point on the secp256k1 curve"

Reply:
Thanks! Done.